sotodlib/tod_ops/flags.py

```diff
@@ -672,3 +692,151 @@ def get_inv_var_flags(aman, signal_name='signal', nsigma=5,
+        right = aman.flags.right_scan.ranges()
+        start_left = 0 if (left[0,0] < right[0,0]) else 1
+        ss_ind = np.empty((left.shape[0] + right.shape[0], 2), dtype=left.dtype)
+        ss_ind[start_left::2] = left
+        ss_ind[(start_left-1)%2::2] = right
+
+    start_inds, end_inds = ss_ind.T
+    n_subscan = ss_ind.shape[0]
+    tt = aman.timestamps
+    subscan_aman = core.AxisManager(aman.samps, core.IndexAxis("subscans", n_subscan))
+
+    is_left = aman.flags.left_scan.mask()[start_inds]
+    subscan_aman.wrap('direction', np.array(['left' if is_left[ii] else 'right' for ii in range(n_subscan)]), [(0, 'subscans')])
+
+    subscan_aman.wrap('start_time', tt[start_inds], [(0, 'subscans')])
+    stop_time = aman.timestamps[ss_ind[:-1,1]]
+    last_time = [2*tt[-1] - tt[-2] if end_inds[-1] == tt.size else tt[end_inds[-1]]]
+    subscan_aman.wrap('stop_time', np.concatenate([stop_time, last_time]), [(0, 'subscans')])
+    rm = RangesMatrix([Ranges.from_array(np.atleast_2d(ss), tt.size) for ss in ss_ind])
+    subscan_aman.wrap('subscan_flags', rm, [(0, 'subscans'), (1, 'samps')]) # True in the subscan
+    if merge:
+        name = 'subscan_info'
+        if overwrite and name in aman:
+            aman.move(name, None)
+        aman.wrap(name, subscan_aman)
+    return subscan_aman
+
+def get_subscan_signal(aman, arr, isub=None, trim=False):
+    """
+    Split an array into subscans.
+
+    Parameters
+    ----------
+    aman : AxisManager
+        Input AxisManager.
+    arr : Array
+        Input array.
+    isub : int
+        (Optional). Index of the desired subscan. May also be a list of indices.
+        If None, all are used.
+    """
+    if isinstance(arr, str):
+        arr = aman[arr]
+    if np.isscalar(isub):
+        out = apply_rng(arr, aman.subscan_info.subscan_flags[isub])
+        if trim and out.size == 0:
+            out = None
+    else:
+        if isub is None:
+            isub = range(len(aman.subscan_info.subscan_flags))
+        out = [apply_rng(arr, aman.subscan_info.subscan_flags[ii]) for ii in isub]
+        if trim:
+            out = [x for x in out if x.size > 0]
+
+    return out
+
+
+def apply_rng(arr, rng):
+    """
+    Helper function to get signals on subcsans.
+
+    Parameters
+    ----------
+    arr : Array
+        Array containing the signal. Should have one axis of len (samps)
+    rng : Ranges
+        Ranges object of len (samps) selecting the desired range
+    """
+    if rng.ranges().size == 0:
+        slc = slice(0,0) #  Return an empty array if rng is empty
+    else:
+        slc = slice(*np.squeeze(rng.ranges()))
+    isamps = np.where(np.array(arr.shape) == rng.count)[0][0]
+    ndslice = tuple((slice(None) if ii != isamps else slc for ii in range(arr.ndim)))
+    return arr[ndslice]
+
+def wrap_info(aman, info_aman_name, info, info_names, merge=True):
+    """Wrap info into an aman. Assumed to be (dets,) or (dets, subscans)"""
```

Comment:
This function at first glance seems to be quite general but I think it's somewhat specific to wrapping in the stats info appropriately choosing whether or not to include subscan axes. Can you just call it `wrap_stats` or `wrap_and_check_subscans` or something similarly specific and give the doc string a bit more description.